test/phoenix/router/forward_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule Phoenix.Router.ForwardTest do
+  use ExUnit.Case, async: true
+  use RouterHelper
+
+  defmodule Controller do
```

Comment:
The context is set after you call the router and is completely in the connection. As long as you have the copy of the connection as it entered the controller, you are fine.